tests/unit/test_parsing.py

```diff
@@ -0,0 +1,68 @@
+import pytest
+
+from ogr.parsing import parse_git_repo, RepoUrl
+
+
+@pytest.mark.parametrize(
+    "url,result",
+    [
+        (
+            "https://host.name/namespace/repo",
+            RepoUrl(
+                repo="repo", namespace="namespace", scheme="https", hostname="host.name"
+            ),
+        ),
+        (
+            "https://host.name/namespace/repo.git",
+            RepoUrl(
+                repo="repo", namespace="namespace", scheme="https", hostname="host.name"
+            ),
+        ),
+        (
+            "http://host.name/namespace/repo",
+            RepoUrl(
+                repo="repo", namespace="namespace", scheme="http", hostname="host.name"
+            ),
+        ),
+        (
+            "git://host.name/namespace/repo",
+            RepoUrl(
+                repo="repo", namespace="namespace", scheme="git", hostname="host.name"
+            ),
+        ),
+        (
+            "git+https://host.name/namespace/repo",
+            RepoUrl(
+                repo="repo",
+                namespace="namespace",
+                scheme="git+https",
+                hostname="host.name",
+            ),
+        ),
+        (
+            "git@host.name:namespace/repo",
+            RepoUrl(
+                repo="repo",
+                namespace="namespace",
+                scheme="http",
```

Comment:
is this ok?

Reply:
I was not sure about that either, but leave that as it was originally in Betka.

For the purpose of OGR, we always need the `http`/`https` for calling the APIs. What about using `git` and force the `http`/`https` in the service classes?

Reply:
`git` = `http`, can we enforce https?